src/user/user.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@nestjs/common'
+
+import { hash } from '../common/utils/crypto.utils'
+import { CreateUserDto } from './dto/create-user.dto'
+import { UpdateUserDto } from './dto/update-user.dto'
+import { UserDatabaseService } from './user.database.service'
+import type { User } from './schemas/user.schema'
+
+// TODO: return all user data without passwords?
```

Comment:
Helper reference from our conversation
![image](https://github.com/user-attachments/assets/d81ff599-0b08-4da5-bc8d-1a3cc3b7d7be)

Reply:
I decided to remove password by returning all fields except `user.password` like it was before, because of `The operand of a 'delete' operator must be optional.` warning:
```
const user = await this.userService.getUserByCredentials(
        username,
        password,
      )
      const { password: _password, ...userWithoutPassword } = user
      return userWithoutPassword
```
Ofc, I can manage types or create new object with new types, but I think this solution is quite sufficient here.